lib/auth/oauth2client.ts

```diff
@@ -0,0 +1,586 @@
+/**
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import AuthClient from './authclient';
+import LoginTicket from './loginticket';
+import PemVerifier from './../pemverifier';
+import { merge } from 'lodash';
+import * as querystring from 'querystring';
+import * as util from 'util';
+const noop = Function.prototype;
+
+export default class OAuth2Client extends AuthClient {
+
+  public _clientId: string;
```

Comment:
I think it is fine to leave for the initial translation but it would be good to do something else longer term.